### PR TITLE
Add a new mode for drawing a circle

### PR DESCRIPTION
**Pull request by Haixiang6123**
## Summary

I just make a new circle mode (ScaleCircleMode) for drawing a circle by moving cursor. Here's the gif demo:

Demo

### CODE REVIEW

**Comment by Haixiang6123** [comment]
How it works:
1. User first clicks anywhere in the map
2. Then user moves his cursor to anywhere, then a circle will be displayed
3. When user clicks for the second time, the circle will be fixed and can be changed

File: __tests__/modes/ScaleCircleMode.test.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  "geometry": {
+    "type": "Polygon",
+    "coordinates": []
+  }
+};
+const doubleClickZoom = require('@mapbox/mapbox-gl-draw/src/lib/double_click_zoom');
+const Constants = require('@mapbox/mapbox-gl-draw/src/constants');
+const circle = require('@turf/circle');
+
+describe('CircleMode tests', () => {
```
Comment (iamanvesh) [inline __tests__/modes/ScaleCircleMode.test.js]: Rename the mode to ScaleCircleMode

File: lib/modes/ScaleCircleMode.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+const doubleClickZoom = require('@mapbox/mapbox-gl-draw/src/lib/double_click_zoom');
+
+const turf = require('@turf/turf');
+
+const ScaleCircleMode = MapboxDraw.modes.draw_polygon;
+
+ScaleCircleMode.onSetup = function () {
+    const polygon = this.newFeature({
+        type: Constants.geojsonTypes.FEATURE,
+        properties: {
```
Comment (iamanvesh) [inline lib/modes/ScaleCircleMode.js]: add the radius too, to the properties.

File: lib/modes/ScaleCircleMode.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    if (state.currentVertexPosition === 0) {
+        return
+    }
+    // Get key points
+    const currentPoint = turf.point([e.lngLat.lng, e.lngLat.lat]);
+    const startPoint = state.startPoint;
+
+    // Compute circle attributes
+    const radius = turf.distance(startPoint, currentPoint) / 2;
+    const midPoint = turf.midpoint(startPoint, currentPoint);
```
Comment (iamanvesh) [inline lib/modes/ScaleCircleMode.js]: usability issue: better to have midpoint as the center of the circle  and radius as just the distance between the current point and the start point

File: lib/modes/ScaleCircleMode.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    if (radius === 0) {
+        return;
+    }
+
+    // Make a circle
+    const circleFeature = turf.circle(midPoint.geometry.coordinates, radius);
+
+    // Reset circle
+    state.polygon.incomingCoords(circleFeature.geometry.coordinates);
+    state.polygon.properties.center = midPoint.geometry.coordinates;
```
Comment (iamanvesh) [inline lib/modes/ScaleCircleMode.js]: compute the radius in kms and add it to the properties

File: lib/modes/ScaleCircleMode.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+    return {
+        polygon,
+        currentVertexPosition: 0
+    };
+};
+
+ScaleCircleMode.onClick = function (state, e) {
+    // First click
+    if (state.currentVertexPosition === 0) {
```
Comment (iamanvesh) [inline lib/modes/ScaleCircleMode.js]: usability: better to add this in onMouseDown

File: lib/modes/ScaleCircleMode.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        state.currentVertexPosition++;
+        state.startLngLat = e.lngLat;
+        state.startPoint = turf.point([e.lngLat.lng, e.lngLat.lat]);
+    }
+    // Second click
+    else {
+        return this.changeMode(Constants.modes.SIMPLE_SELECT, {featureIds: [state.polygon.id]});
```
Comment (iamanvesh) [inline lib/modes/ScaleCircleMode.js]: move this to `onMouseUp`

File: lib/modes/ScaleCircleMode.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    }
+};
+
+ScaleCircleMode.onMouseMove = function (state, e) {
+    if (state.currentVertexPosition === 0) {
```
Comment (iamanvesh) [inline lib/modes/ScaleCircleMode.js]: move all this processing to dragMove and touchmove

File: package.json [hunk 8]
```diff
@@ -22,7 +22,8 @@
     "@turf/circle": "^6.0.1",
     "@turf/distance": "^6.0.1",
     "@turf/helpers": "^6.1.4",
-    "@turf/length": "^6.0.2"
+    "@turf/length": "^6.0.2",
+    "@turf/turf": "^5.1.6"
```
Comment (iamanvesh) [inline package.json]: import the modules that are necessary, turf is a huge library which increases the build size significantly. revert the `package-lock.json` file and the `package.json` before adding the dependencies.

**Pull request closed without merging**